src/tex/triangulation.tex

```diff
@@ -988,28 +988,68 @@ \subsection{General composition of nested and non manifold cycles}
 \paragraph{From a non-contractible polygon to a list of triangles}
 %-------------------------------------------------------------------------------
 @D From a non-contractible polygon to a list of triangles 
-@{""" From a non-contractible polygon to a list of triangles """
-def cycles2triangles(polygon): 
-    triangleSet,triangledFace = [],[]
+@{#""" From a non-contractible polygon to a list of triangles """
+#def cycles2triangles(polygon): 
+#    triangleSet,triangledFace = [],[]
+#    externalCycle = polygon[0]
+#    triangledCycle = []
+#    polyline = []
+#    for p in externalCycle:
+#        polyline.append(Point(p[0],p[1]))
+#    cdt = CDT(polyline)
+#
+#    internalCycles = polygon[1:]
+#    for cycle in internalCycles:
+#        hole = []
+#        for p in cycle:
+#            hole.append(Point(p[0],p[1]))
+#        cdt.add_hole(hole)
+#
+#    triangles = cdt.triangulate()
+#    trias = [ [[t.a.x,t.a.y,0],[t.c.x,t.c.y,0],[t.b.x,t.b.y,0]] 
+#        for t in triangles ]
+#    
+#    triangleSet += [AA(REVERSE)(trias)]
+#    return triangleSet
+
+""" From a non-contractible polygon to a list of triangles """
+from triangle import triangulate
+from architectural import face2edge
+
+def cycles2triangles(polygon):
+    triangleSet = []
+
+    vertices = []
```

Comment:
Un piccolo appunto, se i dati in questo caso sono molti (10^6 o +) Questo doppio for con in può diventare un collo di bottiglia.

`import time

a = range(1,10000000)
aset = set(a)

# Primo Caso
start = time.time()
print(10000000 not in a)
end = time.time()
print(end - start)

# Secondo caso
start = time.time()
print(10000000 not in aset)
end = time.time()
print(end - start)

print("Done")
`

Il primo caso c'impiega **0.173789024353s** il secondo **0.00114893913269s**
Quindi se i dati sono tanti potrebbe essere sensato usare un set() e poi portarlo a [] (usando `list(variabile_set)`)
 ... se  è necessario mantenere l'ordine di inserimento, mantieni sia il vettore sia il set ed inserisci su entrambi ma il check `not in` lo fai sul set.


Se i dati previsti sono pochi, nessun problema :)

Reply:
grazie :)